benchmark/portblas/blas3/gemm_batched.cpp

```diff
@@ -239,13 +239,227 @@ void register_benchmark(blas_benchmark::Args& args,
-      sb_handle_ptr, success, blas_benchmark::utils::MEM_TYPE_BUFFER, gemm_batched_params);
+      sb_handle_ptr, success, blas_benchmark::utils::MEM_TYPE_BUFFER,
+      gemm_batched_params);
 #ifdef SB_ENABLE_USM
   register_benchmark<scalar_t, blas::helper::AllocType::usm>(
-      sb_handle_ptr, success, blas_benchmark::utils::MEM_TYPE_USM, gemm_batched_params);
+      sb_handle_ptr, success, blas_benchmark::utils::MEM_TYPE_USM,
+      gemm_batched_params);
 #endif
 }
 
+#ifdef BLAS_ENABLE_COMPLEX
+template <typename scalar_t, blas::helper::AllocType mem_alloc>
+void run(benchmark::State& state, blas::SB_Handle* sb_handle_ptr, int t1,
+         int t2, index_t m, index_t k, index_t n, std::complex<scalar_t> alpha,
+         std::complex<scalar_t> beta, index_t batch_size, int batch_type_i,
+         bool* success) {
+  // initialize the state label
+  blas_benchmark::utils::set_benchmark_label<std::complex<scalar_t>>(
+      state, sb_handle_ptr->get_queue());
+
+  // Standard test setup.
+  std::string t1s = blas_benchmark::utils::from_transpose_enum(
+      static_cast<blas_benchmark::utils::Transposition>(t1));
+  std::string t2s = blas_benchmark::utils::from_transpose_enum(
+      static_cast<blas_benchmark::utils::Transposition>(t2));
+  const char* t_a = t1s.c_str();
+  const char* t_b = t2s.c_str();
+  auto batch_type = static_cast<blas::gemm_batch_type_t>(batch_type_i);
+
+  index_t lda = t_a[0] == 'n' ? m : k;
+  index_t ldb = t_b[0] == 'n' ? k : n;
+  index_t ldc = m;
+
+  blas_benchmark::utils::init_level_3_cplx_counters<
+      blas_benchmark::utils::Level3Op::gemm_batched, scalar_t>(
+      state, beta, m, n, k, batch_size);
+
+  blas::SB_Handle& sb_handle = *sb_handle_ptr;
+  auto q = sb_handle.get_queue();
+
+  // Matrices
+  std::vector<std::complex<scalar_t>> a =
+      blas_benchmark::utils::random_cplx_data<scalar_t>(m * k * batch_size);
+  std::vector<std::complex<scalar_t>> b =
+      blas_benchmark::utils::random_cplx_data<scalar_t>(k * n * batch_size);
+  std::vector<std::complex<scalar_t>> c =
+      blas_benchmark::utils::const_cplx_data<scalar_t>(m * n * batch_size,
+                                                       scalar_t(0));
+
+#ifdef BLAS_VERIFY_BENCHMARK
+  // Run a first time with a verification of the results
+  std::vector<std::complex<scalar_t>> c_ref = c;
+  auto _base = [=](index_t dim0, index_t dim1, index_t idx) {
+    return dim0 * dim1 * idx;
+  };
+  for (int batch_idx = 0; batch_idx < batch_size; batch_idx++) {
+    reference_blas::cgemm<scalar_t>(
+        t_a, t_b, m, n, k, reinterpret_cast<const void*>(&alpha),
+        reinterpret_cast<const void*>(a.data() + _base(m, k, batch_idx)), lda,
+        reinterpret_cast<const void*>(b.data() + _base(k, n, batch_idx)), ldb,
+        reinterpret_cast<const void*>(&beta),
+        reinterpret_cast<void*>(c_ref.data() + _base(m, n, batch_idx)), ldc);
+  }
+
+  if (batch_type == blas::gemm_batch_type_t::interleaved) {
```

Comment:
I think it's best if we move this check before the `#ifdef BLAS_VERIFY_BENCHMARK` block to avoid calling the reference implementation.